packages/angular_devkit/build_angular/src/babel/webpack-loader.ts

```diff
@@ -144,5 +148,19 @@ export default custom<AngularCustomOptions>(() => {
         ],
       };
     },
+    result(result, { source, map: inputSourceMap }) {
+      if (result.map && inputSourceMap) {
+        result.map = new SourceMapSource(
+          result.code || '',
+          this.resourcePath,
+          result.map,
+          source,
+          inputSourceMap as string,
+          true,
+        ).map();
+      }
+
+      return result;
+    },
```

Comment:
This didn't help in my scenario. But this did:

```js
        result(result, { map: inputSourceMap }) {
            if (result.map && inputSourceMap) {
              const fs = require('fs');
              const path = require('path');
              const sourcemaps = require('@angular/compiler-cli/src/ngtsc/sourcemaps');

              const fileSystem = {
                resolve: path.resolve,
                exists: fs.existsSync,
                dirname: path.dirname,
                relative: path.relative,
                readFile: fs.readFileSync,
              };
              const loader = new sourcemaps.SourceFileLoader(fileSystem, {}, {});
              const outputSource = loader.loadSourceFile(this.resourcePath, result.code, {map: result.map, path: this.resourcePath + '.map'});
              result.map = outputSource.renderFlattenedSourceMap()
            }
            return result;
        },
```

Reply:
But this is quite resource intensive and ends up running for a large number of files, not just the ones that have been linked.